packages/nimble-components/src/wafer-map/specs/README.md

```diff
@@ -0,0 +1,199 @@
+We are planning to perform initial benchmarking on this aspect, but not planning to optimize it in the first release.
+### Prior Art/Examples
+
+Please see below a few screenshots regarding the current `wafer-map` component:
+
+![WaferMap Example](file:///Resources/wafer_map_example.png)
+
+## Design
+
+From a user’s perspective one of the most important actions which this component should be able to achieve is to zoom in and out on the datapoints. This functionality should occur when the mouse pointer hovers the `nimble-wafer-map` component and a mouse wheel action is performed.
+While zooming in on the component, the user should also be able to pan (drag) the wafer map by holding the left mouse button and moving the cursor on the XY axis.
+
+From a developer perspective, it's very important that the `nimble-wafer-map` component can be integrated with the MicroStrategy Custom plugin framework as this framework provides functionalities as follows:
+* Tooltip when hovering any if the datapoints withing the wafer map
+* Wafer map legend
+* MicroStrategy specific context menus on right-click action
+
+Information regarding the basic MicrosStrategy **Custom Plugin Workflow** can be found [here](https://www2.microstrategy.com/producthelp/Current/VisSDK/Content/topics/HTML5/Understanding_the_workflow.htm)
+Also, the visualisation widget has no active controls, so there is no way to alter the visual aspect or data post rendering. Every input parameter must be provided through the constructor of the wafer map component.
+
+### API
+
+*The key elements of the component's public API surface:*
+* Component Name: `nimble-wafer-map`
+* Props/Attrs: 
+  * `die[]` - this represents the input data, which fills the wafer map with content.\
+  The **die**  object contains the following attributes:
+	* XCoorrdianate: integer
+	* YCoordinate: integer
+	* Value: float
+  * `quadrant` - represents the orientation fo the dies on the wafer map (the layout of the values on the dies). It can be represented by an Enum with the following values: 
+	* TopLeft - ![Top Left Quadrant](file:///Resources/top_left.png)
+	* BottomLeft - ![Bottom Left Quadrant](file:///Resources/bottom_left.png)
+	* TopRight - ![Top Right Quadrant](file:///Resources/top_right.png)
+	* BottomRight - ![Bottom Right Quadrant](file:///Resources/bottom_right.png)
+  * `orientation` - represent the orientation of the notch on the wafer map outline (only visual). As only four static orientations are possible, it can be represented by an Enum with the following values: top, bottom, left, right.
+  * `color_scale` - represents the color spectrum which shows the status of the dies on the wafer.
+  * `number_of_characters` - represents the number of characters allowed to be displayed within a single die. As the die values are represented by Floating point numbers, we must have the liberty of limiting how many characters we are willing to display within a single die.
```

Comment:
I'd propose a more specific name that explains which text this is controlling. Maybe `dieCharacterCount`? `maxDieLabelLength`?